### PR TITLE
refactor: intersection module

### PR DESCRIPTION
**Pull request by soblin**
## Description

refactored the `modifyPathVelocity` function to

```
initializeRTCStatus()
modifyPathVelociyDetail()
prepareRTCStatus()
reactRTCApproval()
```

parts

## Related links

[COMPANY INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-2639)

[COMPANY INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-2581)

## Tests performed

- scenario
  - https://evaluation.tier4.jp/evaluation/reports/6c007748-55b1-538e-9547-e884c1e4f1ff?project_id=prd_jt
  - https://evaluation.tier4.jp/evaluation/reports/f672fed4-651d-57c9-8fb6-4e080a3082a6?project_id=prd_jt
 
There were not degraded scenarios.

## Notes for reviewers

parameter PR: https://github.com/autowarefoundation/autoware_launch/pull/391

## Interface changes

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_intersection_module/README.md [hunk 1]
```diff
@@ -2,120 +2,133 @@
...
 
-Target objects are limited by lanelets to prevent unexpected behavior. A lane that satisfies the following conditions is defined as an "Attention Lane" and used to define the target object.
+#### Right Of Way
 
-- The lane crosses with the driving lane of the ego-vehicle
-- The lane has high priority for the driving lane of the ego-vehicle (priority tags are needed to be configured in Lanelet-map according to the situation).
+Following table shows an example of how to assign `right_of_way` tag and set `yield_lanes` to each lane in intersections.
 
-See the following figures to know how to create an attention area and its rationale.
+| turn direction \ traffic light | w/ traffic light                                                | w/o traffic light                                |
```
Comment (shmpwk) [inline planning/behavior_velocity_intersection_module/README.md:37]:
```suggestion
| turn direction / traffic light | w/ traffic light                                                | w/o traffic light                                |
```

File: planning/behavior_velocity_intersection_module/src/manager.cpp [hunk 2]
```diff
@@ -65,10 +63,12 @@ IntersectionModuleManager::IntersectionModuleManager(rclcpp::Node & node)
     vehicle_info.max_longitudinal_offset_m;
   ip.stuck_vehicle.stuck_vehicle_vel_thr =
     node.declare_parameter<double>(ns + ".stuck_vehicle.stuck_vehicle_vel_thr");
+  /*
   ip.stuck_vehicle.assumed_front_car_decel =
     node.declare_parameter<double>(ns + ".stuck_vehicle.assumed_front_car_decel");
   ip.stuck_vehicle.enable_front_car_decel_prediction =
     node.declare_parameter<bool>(ns + ".stuck_vehicle.enable_front_car_decel_prediction");
+  */
```
Comment (shmpwk) [inline planning/behavior_velocity_intersection_module/src/manager.cpp:71]: Do you mean it is still in progress or is it intended?

Comment (soblin) [reply planning/behavior_velocity_intersection_module/src/manager.cpp:71]: This part is not used as of now, so I commented out temporarily.

**Review by shmpwk — APPROVED** [review]
I checked the ego behavior on intersection.
The virtual walls of intersection and intersection occlusion behave as I expected.
The code also looks good to me.

**Pull request merged**